app/routes.py

```diff
@@ -1,2 +1,274 @@
+def handle_tasks():
+    if request.method == "GET":
+        sort_query = request.args.get("sort")
+        if sort_query == "asc":
+            tasks = Task.query.order_by(Task.title)
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(desc(Task.title))
+        else:
+            tasks = Task.query.all()
+
+        tasks_response = []
+        for task in tasks:
+            if not task.completed_at:
+                is_complete = False
+            else:
+                is_complete = True
+            tasks_response.append({
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": is_complete
+            })
+        return make_response(jsonify(tasks_response), 200)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        keys = ['title', 'description', 'completed_at']
+        for key in keys:
+            if key not in request_body:
+                task_response = {"details": "Invalid data"}
+                return make_response(task_response, 400)
+
+        new_task = Task(title=request_body["title"],
+                        description=request_body["description"],
+                        completed_at=request_body["completed_at"])
+
+        db.session.add(new_task)
+        db.session.commit()
+        if not request_body["completed_at"]:
+            is_complete = False
+        else:
+            is_complete = True
+        task_response = {"task":
+                         {"id": new_task.task_id,
+                          "title": new_task.title,
+                          "description": new_task.description,
+                          "is_complete": is_complete}
+                         }
+        return make_response(task_response, 201)
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+    if not task:
+        return make_response(f"Task #{task_id} Not Found", 404)
+
+    if request.method == "GET":
+        if not task.completed_at:
+            is_complete = False
+        else:
+            is_complete = True
+        task_response = {"task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": is_complete,
+        }}
+        if task.goal_id:
+            task_response["task"]["goal_id"] = task.goal_id
+        return make_response(task_response, 200)
+    elif request.method == "PUT":
+        form_data = request.get_json()
+        if ("completed_at" not in form_data or
+                form_data["completed_at"] == None):
+            is_complete = False
+        else:
+            is_complete = True
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+        task.completed_at = form_data["completed_at"]
```

Comment:
I mentioned this already above, but we should be sure that the same fields required for POSTing are included here for PUT. PUT _replaces_ the value for the supplied task id, so we should ensure that all of the values required to represent a `Task` are provided with the same restrictions as we had when creating it in the first place.